gui/wxpython/core/watchdog.py

```diff
@@ -0,0 +1,219 @@
+                                                            hold object instance
+                                                            of core.gconsole.GConsole
+                                                            class
+    :param str patterns: map watchdog patterns with default value "*" all
+    """
+
+    def __init__(self, elements_dirs, evt_handler, gconsole, patterns="*"):
+        self._elements_dirs = elements_dirs
+        self._evt_handler = evt_handler
+        self._patterns = patterns
+        self._gconsole = gconsole
+        self._observer = None
+
+    def _getGconsole(self):
+        """Lazy loading of object instance of core.gconsole.GConsole class
+
+        :return object instance: object instance of class
+                                 core.gconsole.GConsole
+        """
+        gconsole = "_gconsole"
+        if hasattr(self._gconsole, gconsole):
+            return getattr(self._gconsole, gconsole)
+        return self._gconsole
+
+    def ScheduleWatchCurrentMapset(self):
+        """Using watchdog library, sets up watching of current mapset folder
+        to detect changes not captured by other means (e.g. from command line).
+        Schedules 1 watches (raster).
+        If watchdog observers are active, it restarts the observers in
+        current mapset.
+        """
+        global watchdog_used
+        if not watchdog_used:
+            return
+
+        if self._observer and self._observer.is_alive():
+            self._observer.stop()
+            self._observer.join()
+            self._observer.unschedule_all()
+        self._observer = Observer()
+
+        gisenv = grass.gisenv()
+        mapset_path = os.path.join(
+            gisenv["GISDBASE"], gisenv["LOCATION_NAME"], gisenv["MAPSET"]
+        )
+        rcfile = os.environ["GISRC"]
+        self._observer.schedule(
+            CurrentMapsetWatch(rcfile, mapset_path, self._evt_handler),
+            os.path.dirname(rcfile),
+            recursive=False,
+        )
+        for element, directory in self._elements_dirs:
+            path = os.path.join(mapset_path, directory)
+            if not os.path.exists(path):
+                try:
+                    os.mkdir(path)
+                except OSError:
+                    pass
+            if os.path.exists(path):
+                self._observer.schedule(
+                    MapWatch(self._patterns, element, self._evt_handler),
+                    path=path,
+                    recursive=False,
+                )
+        try:
+            self._observer.start()
+        except OSError:
+            # in case inotify on linux exceeds limits
+            watchdog_used = False
+            self._getGconsole().WriteWarning(
+                _(
+                    "File size limit exceeded. The current mapset"
+                    " and maps watchdog are disabled now."
+                ),
+            )
+            return
```

Comment:
Could you pass giface and use giface.WriteWarning here?

Reply:
Yes of course, thanks.